app/models/form.rb

```diff
@@ -93,6 +95,13 @@ def make_unlive!
 
 private
 
+  def set_external_id
+    loop do
+      self.external_id = SecureRandom.uuid[0, 8]
```

Comment:
Any reason not to use `SecureRandom.hex(8)`?

https://ruby-doc.org/3.2.2/stdlibs/random/Random/Formatter.html#method-i-hex

Reply:
Woops, I misread the docs, I meant `SecureRandom.hex(4)` 😁